doc/NAT/SAI-NAT-API.md

```diff
@@ -0,0 +1,297 @@
+```sh
+sai_nat_counter_list[0].id = SAI_NAT_COUNTER_ATTR_NAT_TYPE;
+sai_nat_counter_list[0].value.u32 = SAI_NAT_SOURCE_NAT;
+
+sai_nat_counter_list[1].id = SAI_NAT_COUNTER_ATTRZONE_ID;
+sai_nat_counter_list[1].value.u8 = 100; 
+
+sai_nat_counter_list[2].id = SAI_NAT_COUNTER_ATTR_ENABLE_DISCARD;
+sai_nat_counter_list[2].value = true;
+sai_nat_counter_list[3].id = SAI_NAT_COUNTER_ATTR_ENABLE_TRANSLATION_NEEDED;
+sai_nat_counter_list[3].value = true;
+
+sai_nat_counter_list[4].id = SAI_NAT_COUNTER_ATTR_ENABLE_TRANSLATIONS;
+sai_nat_counter_list[4].value = true;
+
+nat_attr_count = 5;
+sai_create_nat_counter_fn(
+    &sai_src_nat_counter_obj,
+    switch_id,
+    nat_attr_count,
+    sai_nat_attr_list);
+```
+
+##### Step 3.2: Create a SNAT object:
+```sh
+sai_nat_attr_list[0].id = SAI_NAT_ATTR_NAT_TYPE;
+sai_nat_attr_list[0].value.u32 = SAI_NAT_SOURCE_NAT;
+
+sai_nat_attr_list[1].id = SAI_NAT_ATTR_FROM_ZONE_ID;
+sai_nat_attr_list[1].value.u8 = 100; 
+
+sai_nat_attr_list[2].id = SAI_NAT_ATTR_TO_ZONE_ID;
+sai_nat_attr_list[2].value.u8 = 200; 
+
+sai_nat_attr_list[3].id = SAI_NAT_ATTR_NAT_ENTRY_LIST;
+sai_nat_attr_list[3].value.objectlist.count = 1;
+sai_nat_attr_list[3].value.objectlist.list[0] = sai_src_nat_entry_obj; 
+
+sai_nat_attr_list[4].id = SAI_NAT_ATTR_NAT_COUNTER_LIST;
+sai_nat_attr_list[4].value.objectlist.count = 1;
+sai_nat_attr_list[4].value.objectlist.list[0] = sai_src_nat_counter_obj; 
+
+nat_attr_count = 5;
+sai_create_nat_fn(
+    &sai_src_nat_obj,
+    switch_id,
+    nat_attr_count,
+    sai_nat_attr_list);
+```
+
+##### Step 4: Create a DNAT object:
+```sh
+sai_nat_attr_list[0].id = SAI_NAT_ATTR_NAT_TYPE;
+sai_nat_attr_list[0].value.u32 = SAI_NAT_TYPE_DESTINATION_NAT;
+
+sai_nat_attr_list[1].id = SAI_NAT_ATTR_FROM_ZONE_ID;
+sai_nat_attr_list[1].value.u8 = 200; 
+
+sai_nat_attr_list[2].id = SAI_NAT_ATTR_TO_ZONE_ID;
+sai_nat_attr_list[2].value.u8 = 100; 
+
+sai_nat_attr_list[3].id = SAI_NAT_ATTR_NAT_ENTRY_LIST;
+sai_nat_attr_list[3].value.objectlist.count = 1; 
+sai_nat_attr_list[3].value.objectlist.list[0] = sai_dst_nat_entry_obj; 
+
+nat_attr_count = 4;
+    sai_create_nat_fn(
+    &sai_dst_nat_obj,
+    switch_id,
+    nat_attr_count,
+    sai_nat_attr_list);
+```
+## NAT Per flow and Zone Counters
+Following per zone statistics is provided by the hardware for NAT feature.
+- DNAT_DISACARDS/SNAT_DISCARDS – If Packet is not TCP/UDP and/or is a fragmentated IP packet. 
+- DNAT_TRANSLATION_NEEDED/SNAT_TRANSLATION_NEEDED – If there is NAT table lookup miss for TCP/UDP packets, then this counter is incremented.
+- DNAT_TRANSLATIONS/SNAT_TRANSLATIONS – If NAT table lookup is a hit, then this counter is incremented.
+
+Counters can be attached to per NAT entry to gather per flow statistics. Per NAT entry stat is enabled in the entry attributes.
+
+> Per zone counters are created using sai_create_nat_counter_fn() API.
+
+NAT counters can be extracted using TAM streaming interface or by using TAM GET API.

```

Comment:
During the TAM review, there was a consensus that there should be a GET API to get the data and same data can be streamed as well. 
From SONiC implementation perspective this becomes very simple single API invocation.

Reply:
My comment was not referring to the TAM get API
I was referring to the fact we should be able to query the statistics as any other statistics in SAI, in a standard get_<objects>_stats_fn

Reply:
Yes, there was a consensus then instead of having a proliferation of get_stats API and then NOS parsing output of each API, can we have simpler API design hence the usage of GET API.
Think of it as an IOCTL for stats